dags/utils/amplitude.py

```diff
@@ -0,0 +1,133 @@
+from airflow import models
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow.contrib.operators.bigquery_operator import BigQueryOperator
+from airflow.contrib.operators.bigquery_to_gcs import BigQueryToCloudStorageOperator
+from airflow.contrib.operators.gcs_to_s3 import GoogleCloudStorageToS3Operator
+from operators.bq_sensor import BigQuerySQLSensorOperator
+from operators.gcs import GoogleCloudStorageDeleteOperator
+from os import environ
+
+
+def export_to_amplitude(
+        parent_dag_name,
+        dag_name,
+        default_args,
+        dataset,
+        table_or_view,
+        s3_prefix,
+        gcs_bucket='moz-fx-data-derived-datasets-amplitude-export',
+        gcp_conn_id='google_cloud_derived_datasets',
+        amplitude_s3_conn='amplitude_s3_conn',
+        amplitude_s3_bucket='com-amplitude-vacuum-mozilla-vacuum-wup'):
+
+    """ Export a bigquery table or view to Amplitude.
+
+    This uses the BigQueryToCloudStorage operator to export the
+    partition to GCS, then pushes that data to S3. It operates
+    on a temporary table that is dropped after the job is finished.
+
+    :param str parent_dag_name: Parent dag name
+    :param str dag_name: This dag's name (appended to parent_dag_name)
+    :param str default_args: DAG configuration
+    :param str dataset: BigQuery dataset
+    :param str table_or_view: Table or view name
+    :param str gcs_bucket: The bucket the data will be exported to
+    :param str gcp_conn_id: GCP connection ID
+    :param str amplitude_s3_conn: S3 connection ID
+    :param str amplitude_s3_bucket: The bucket to export data to
+    :param str s3_prefix: The prefix for the s3 objects
+    """
+
+    environment = environ['DEPLOY_ENVIRONMENT']
+    _dag_name = '{}.{}'.format(parent_dag_name, dag_name)
+
+    with models.DAG(_dag_name, default_args=default_args) as dag:
+        # For now, we assume the view is already updated
+        # See https://github.com/mozilla/bigquery-etl/issues/218
+
+        exec_date = '{{ ds }}'
+
+        # Check that we have data for this date
+        check_sql = (
+            'SELECT COUNT(*) '
+            'FROM {}.{} '
+            'WHERE DATE(submission_timestamp) = "{}"'
+        ).format(dataset, table_or_view, exec_date)
+
+        wait_for_data = BigQuerySQLSensorOperator(
```

Comment:
Instead of implementing this sensor can we use the airflow operator sensor that we use in several other DAGs already? I'm assuming the underlying data is populated in some other DAG, so we can check for the relevant task that populates the underlying table here rather than querying BQ.

Reply:
Unfortunately, this job doesn't know anything about where the upstream data is being created, or we would. Additionally this is a view, so we need to check that table that the view is reading from, which is not readily available.